assets/performanceprofile/scripts/cpuset-configure.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# cpuset-configure.sh configures three cpusets in preparation for allowing containers to have cpu load balancing disabled.
+# To configure a cpuset to have load balance disabled (on cgroup v1), a cpuset cgroup must have `cpuset.sched_load_balance`
+# set to 0 (disable), and any cpuset that contains the same set as `cpuset.cpus` must also have `cpuset.sched_load_balance` set to disabled.
+
+set -euo pipefail
+
+root=/sys/fs/cgroup/cpuset
+system="$root"/system
+machine="$root"/machine.slice
+
+# As such, the root cgroup needs to have cpuset.sched_load_balance=0. 
+echo 0 > "$root"/cpuset.sched_load_balance
+
+# However, this would present a problem for system daemons, which should have load balancing enabled.
+# As such, a second cpuset must be created, here dubbed `system`, which will take all system daemons.
+# Since systemd starts its children with the cpuset it is in, moving systemd will ensure all processes systemd begins will be in the correct cgroup.
+mkdir "$system"
+# cpuset.mems must be initialized or processes will fail to be moved into it.
+cat "$root/cpuset.mems" > "$system"/cpuset.mems
+# Retrieve the cpuset of systemd, and write it to cpuset.cpus of the system cgroup.
+reserved_set=$(taskset -cp  1  | awk 'NF{ print $NF }')
+echo "$reserved_set" > "$system"/cpuset.cpus
+
+# And move the system processes into it.
+# Note, some kernel threads will fail to be moved with "Invalid Argument". This should be ignored.
+for process in $(cat "$root"/cgroup.procs | sort -r); do
+	echo $process > "$system"/cgroup.procs 2>&1 | grep -v "Invalid Argument" || true;
+done
+
+# Finally, a the `machine.slice` cgroup must be preconfigured. Podman will create containers and move them into the `machine.slice`, but there's
```

Comment:
How is podman using the machine slice? Is that a workload partitioning specialty?

Reply:
podman puts all of its systemd cgroup containers into the machine.slice by default. it was formally not at all configured for workload partitioning because the podman pieces run before guaranteed pods are in the picture. The slice remains on the node only as a remnant because nothing removes them